### PR TITLE
fix(attributes): escape commas in a definition's alias (closes #6853)

### LINKED ISSUE #6853: Aliases for Label/Relation definitions should either forbit the character `,` or escape it
### Description

From both the GUI and via text-based attribute definition, it is possible to set an Alias that contains a coma:
Image

This comes up as problematic because that character is used as a separator to describe the attribute itself.

While the long-term plan is to improve upon that through #6421 , it would probably help in the meantime to catch this case through validation, and escape it (e.g. `#label:myLabel="promoted,alias='myLabel, very pretty',single,text") when typed in the GUI.

### TriliumNext Version

0.98.1

### What operating system are you using?

Other Linux

### What is your setup?

Local (no sync)

### Operating System Version

Linux

### Error logs

_No response_

### PR DESCRIPTION
**Pull request by maphew**
## Why

A display name (alias) with a comma breaks a label or relation definition. In the attribute detail dialog, the alias `myLabel, very pretty` is saved, but it shows as `myLabel` after the dialog is opened again. When the text after the comma is a valid token, the token is applied. I ran the `upstream/main` parser on the strings that `serialize()` writes: the alias `Foo, promoted` makes a definition promoted, `Pick, options=x` adds a select option, `Amount, precision=2` sets a precision, and `Parent, inverse=x` sets an inverse relation. `Size, number` does not change the type there, because `serialize()` writes the type after the alias and the last token wins. It does change the type in a hand-typed definition that has the alias last (`promoted,single,text,alias=Size, number` reads as type `number`).

Root cause: a definition is stored as comma-separated tokens. In `packages/commons/src/lib/promoted_attribute_definition_parser.ts`, `parse()` does `value.split(",")` before it looks at any token, and `serialize()` wrote the alias raw (`alias=${definition.promotedAlias}`). Thus `promoted,alias=myLabel, very pretty,single,text` is read as the alias `myLabel` plus a stray token `very pretty`. The dialog (`attribute_detail.tsx`, alias `FormTextBox` to `commitDefinition()` to `serialize()`) does no validation.

Closes #6853

## What

- `serialize()` writes the alias through a new private `encodeAlias()`: `%` to `%25` first, then `,` to `%2C`.
- `parse()` reads the alias through `decodeAlias()`: `%2C` to `,`, then `%25` to `%` last. `alias` with no `=` still gives `undefined`.
- This is the same substitution that `encodeOption()`/`decodeOption()` already use for select options, for the same reason: all released parsers split on `,` first, so the quoted form that the issue suggests (`alias='a, b'`) would still be cut by them. `;` is not encoded, because it separates nothing in an alias.
- No caller changes. Each writer of a user-typed alias uses `serialize()` (`attribute_detail.tsx`, `services/attributes.ts`, the board API, migration 0240) and each reader uses `parse()`.

Trade-offs, the same ones already accepted for select options: (a) a stored alias that literally contains `%2C` or `%25` now reads as `,` or `%`; (b) a client that is not upgraded shows the escaped text (`My%2C Label`, `100%25`), but it writes the same string back, so the data is not lost and there is no write conflict between sync peers; (c) a script that reads the raw definition value sees the escaped alias, as it already does for options.

## Validation

All commands were run in the branch worktree, on the branch head.

- Red, tests added and fix not applied: `pnpm --filter @triliumnext/commons test promoted_attribute_definition_parser` gave `3 failed | 19 passed (22)`. The round trip returned `"promotedAlias": "myLabel"` for `"myLabel, very pretty"`, and `serialize()` returned `alias=myLabel, very pretty,single,text`.
- Green, same command with the fix: `Test Files 1 passed (1)`, `Tests 22 passed (22)`.
- `pnpm typecheck`: `No errors found.`
- `pnpm --filter @triliumnext/client test promoted_attributes.spec PromotedAttributes.spec group_by.spec PromotedAttributesCard.spec`: `Test Files 4 passed (4)`, `Tests 81 passed (81)`.
- `pnpm --filter @triliumnext/commons test builtin_attributes`: 4 passed.
- The new tests do a round trip with `toEqual` on aliases whose fragments are valid tokens (`Size, number`, `Foo, promoted`, `a, multi, options=x`, `Parent, inverse=x`) and on escape edge cases (`100%`, `50%25`, `%2C`, `a=b`, `a;b`), so an applied stray token fails the test.
- Not run: the full test suites, ESLint (the repository rules say not to run it), and the app. I did not do a live check. Manual check: in the attribute detail dialog, clear Promoted on a label definition, set its alias to `A, promoted`, save, then open the dialog again. The alias must be the same and Promoted must stay clear.
- Claude review of the exact diff: verdict ship, no regression found. It listed four gaps that were there before this change and three notes for this text. I made no code change. The gaps are in Scope below.
- Codex cross-vendor review of the exact diff: the `gpt-5.6-sol` run stopped with HTTP 400 (the model is not available on this account) and gave no review. I ran `codex review --base upstream/main` again with `gpt-5.6-terra` at high reasoning effort. It reported no findings.

## Scope

- The ribbon attribute editor stores typed text as it is, so a hand-typed definition must use `%2C` for a comma in the alias. A raw comma there is still split. The issue names this path too, and this change does not add validation to it.
- `buildPromotedDefinition()` in `packages/trilium-core/src/services/import/collection_utils.ts` still replaces `,` and `=` in an imported column name with a space. It can use `serialize()` in a follow-up, but that changes the import output and its tests.
- `hidden_subtree.ts` and `hidden_subtree_templates.ts` write `alias=${t(...)}` raw. No shipped translation of those keys contains a comma now, so this is a latent risk and not a defect.
- `parse()` trims each token, so leading or trailing spaces in an alias are still lost. Select options have the same behavior.
- Docs: `Promoted Attributes.md` in the User Guide shows a hand-typed definition with `alias=Color` and does not mention an escape. Nothing on the page is now wrong. I did not add the `%2C` note, because the User Guide must be edited with `pnpm edit-docs:edit-docs`, and the select option escape is not documented either.
- `encodeAlias()` and `encodeOption()` are almost the same. I kept them apart so that the `;` difference stays visible.

_claude-fable-5-1-high on behalf of matt wilkie_

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: packages/commons/src/lib/promoted_attribute_definition_parser.ts [hunk 1]
```diff
@@ -166,6 +173,21 @@ function decodeOption(option: string): string {
...
+ * choice; `;` separates nothing in an alias, so it is stored verbatim.
+ */
+function encodeAlias(alias: string): string {
+    // `%` first, so the escape written below is not re-escaped.
+    return alias.replace(/%/g, "%25").replace(/,/g, "%2C");
+}
+
+/** The inverse of {@link encodeAlias}: `%25` last, so a decoded `%` cannot start a triplet. */
+function decodeAlias(alias: string): string {
+    return alias.replace(/%2C/gi, ",").replace(/%25/gi, "%");
```
[the remark this answers is not in the document]
Comment (maphew) [reply packages/commons/src/lib/promoted_attribute_definition_parser.ts:188]: This is the trade-off listed under "What" in the PR body, and it is the same one `encodeOption()`/`decodeOption()` already accept for select options in this file (6c19d14a41). A stored alias is affected only if it holds the literal text `%2C` or `%25`. A format marker is not possible without a new token, and released parsers drop tokens they do not know, so an older sync peer would lose the alias instead of showing it escaped. I kept the behavior the same as the existing option escape so the two fields do not differ. If a maintainer prefers a migration or a marker, I can add it.

_claude-fable-5-1-high on behalf of matt wilkie_
